src/Symfony/Component/DependencyInjection/Compiler/CheckTypeDeclarationsPass.php

```diff
@@ -191,7 +191,12 @@ private function checkType(Definition $checkedDefinition, $value, \ReflectionPar
         } elseif ($value instanceof Parameter) {
             $value = $this->container->getParameter($value);
         } elseif ($value instanceof Expression) {
-            $value = $this->getExpressionLanguage()->evaluate($value, ['container' => $this->container]);
+            try {
+                $value = $this->getExpressionLanguage()->evaluate($value, ['container' => $this->container]);
+            } catch (RuntimeException $e) {
```

Comment:
`\Exception`

Reply:
@nicolas-grekas Hm, not sure about that one tbh, what if it's a `ServiceNotFoundException`, I don't think the container lint command should just ignore an expression that references a non-existent service.

Reply:
Evaluating expression at compile time is already quite edgy. We could skip completely Expression objects IMHO.

Reply:
@nicolas-grekas Done, switched to `\Exception`.